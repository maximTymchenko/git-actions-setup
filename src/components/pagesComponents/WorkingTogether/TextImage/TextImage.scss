@import '../../../../styles/vars';

.text-image {
    &__wrapper {
        display: flex;
        align-items: flex-end;
        width: 100%;
        max-width: $container__size - 500px;

        @media screen and (max-width: $mobile) {
            margin-bottom: 0px;
            margin-top: 0;
            display: flex;
        }

        &--centered {
        }
        &--reverse {
            flex-direction: row-reverse;
            align-items: center;
            justify-content: space-between;

            @media screen and (max-width: $tablet) {
                flex-direction: row;
            }
        }
    }

    &__body {
        @media screen and (max-width: $mobile) {
            flex-basis: 100%;
        }

        &--shadow {
            margin-right: 30px;
            position: relative;

            @media screen and (max-width: $mobile) {
                margin-left: 5px;
                margin-top: 25px;
            }

            &::before {
                position: absolute;
                display: block;
                z-index: 20;
                content: '';
                width: 100%;
                // max-width: 590px;
                height: 100%;
                top: 0;
                right: 0;
                left: 0;
                bottom: 0;
                background-color: $primary__light--15;

                @media screen and (max-width: $mobile) {
                    max-width: 360px;
                }

                @media screen and (max-width: $small__mobile) {
                    max-width: 320px;
                }
            }
        }
    }

    &__images {
        display: block;
        height: auto;
        width: auto;
        max-width: 500px;
        // max-height: 497px;
        position: relative;

        @media screen and (max-width: $small__desktop) {
            max-width: 850px;
        }

        @media screen and (max-width: $large__tablet) {
            max-width: 740px;
        }

        @media screen and (max-width: $tablet) {
            max-width: 630px;
        }

        @media screen and (max-width: $small__tablet) {
            max-width: 460px;
        }

        @media screen and (max-width: $small__mobile) {
            max-width: 320px;
        }

        @media screen and (max-width: $mobile) {
            max-width: 360px;
        }

        @media screen and (max-width: $small__mobile) {
            max-width: 300px;
        }

        &--relative {
            right: -100px;
            top: 80px;

            @media screen and (max-width: $small__desktop) {
                left: 120px;
                right: 0;
            }

            @media screen and (max-width: $tablet) {
                left: 10px;
            }

            @media screen and (max-width: $large__tablet) {
                left: 50px;
            }

            @media screen and (max-width: $small__mobile) {
                left: 20px;
            }
        }

        &--dislocation {
            top: 80px;
            right: 0;

            @media screen and (max-width: $desktop) {
                right: -35px;
            }

            @media screen and (max-width: $tablet) {
                right: -167px;
            }

            @media screen and (max-width: $mobile) {
                top: 90px;
                right: -55px;
            }
        }

        &--moved {
            // max-width: 800px;
        }

        &--margin {
            position: static;
            margin-left: 20px;

            @media screen and (max-width: $desktop) {
                margin-left: 20px;
            }

            @media screen and (max-width: $tablet) {
                margin-left: 0px;
            }

            @media screen and (max-width: $mobile) {
                margin-top: 25px;
            }
        }
    }
}
