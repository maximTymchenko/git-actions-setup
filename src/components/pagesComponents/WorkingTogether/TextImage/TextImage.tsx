import React, { FC } from 'react';
import './TextImage.scss';
import SectionHeader from '../../HowWeWork/SectionHeader';

interface ISectionHeader {
    large: string;
    image: any;
    relative?: string;
    typeValue: string;
    size: string;
    position: string;
    shadow?: string;
    id: string;
}
const TextImage: FC = (props: ISectionHeader) => {
    return (
        <div
            data-testid="text-image"
            id={props.id}
            className={`text-image__wrapper text-image__wrapper--${props.position}`}
        >
            <div className={`text-image__body text-image__body--${props.shadow}`}>
                <img
                    className={`text-image__images text-image__images--${props.relative}`}
                    src={props.image}
                    alt=""
                />
            </div>
        </div>
    );
};

export default TextImage;
