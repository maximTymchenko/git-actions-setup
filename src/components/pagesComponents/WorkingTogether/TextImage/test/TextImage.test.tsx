import React from 'react';
import TextImage from '../TextImage';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
it('renders elem correctly', () => {
    const { getByTestId } = render(
        <TextImage
            id="responsibility"
            titleText="Outcome responsibility"
            secondaryTitleText="Blackbird guarantees"
            large="large"
            descriptionMain="You get a wealth of our experience from actively working on miscellaneous client projects that your in-house teams would not probably collectively have. 
                    Regardless, of the hiring model you choose, we will always make sure you are aware of all possibilities and potential pitfalls in advance of making your final decision."
            relative="dislocation"
            typeValue="lighter"
        />,
    );
    expect(getByTestId('text-image')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer
        .create(
            <TextImage
                id="responsibility"
                titleText="Outcome responsibility"
                secondaryTitleText="Blackbird guarantees"
                large="large"
                descriptionMain="You get a wealth of our experience from actively working on miscellaneous client projects that your in-house teams would not probably collectively have. 
                    Regardless, of the hiring model you choose, we will always make sure you are aware of all possibilities and potential pitfalls in advance of making your final decision."
                relative="dislocation"
                typeValue="lighter"
            />,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
