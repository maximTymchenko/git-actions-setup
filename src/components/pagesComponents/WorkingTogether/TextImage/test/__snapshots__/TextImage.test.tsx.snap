// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="text-image__wrapper text-image__wrapper--undefined"
  data-testid="text-image"
  id="responsibility"
>
  <div
    className="text-image__body text-image__body--undefined"
  >
    <img
      alt=""
      className="text-image__images text-image__images--dislocation"
    />
  </div>
</div>
`;
